gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/query_service/api/ApiMetadataQueryServiceImpl.java

```diff
@@ -65,13 +65,14 @@ public Map<String, ApiMetadata> findApiMetadata(String apiId) {
                         (key, existing) ->
                             Optional
                                 .ofNullable(existing)
-                                .map(value -> value.withValue(m.getValue()))
+                                .map(value -> value.withApiId(apiId).withName(m.getName()).withValue(m.getValue()))
```

Comment:
```suggestion
                                .map(value -> value.toBuilder().apiId(apiId).name(m.getName()).value(m.getValue()).build())
```

`@with` will create a new instance. With this 3 `withXXX` we are creating 3 objects for nothing 🙈